lib/kino/smart_cell/chart_builder.ex

```diff
@@ -0,0 +1,225 @@
+defmodule Kino.SmartCell.ChartBuilder do
+  use Kino.JS, assets_path: "lib/assets/chart_builder"
+  use Kino.JS.Live
+  use Kino.SmartCell, name: "Chart builder"
+
+  @as_int ["width", "height"]
+  @as_atom ["data_variable", "chart_type", "x_field_type", "y_field_type", "color_field_type"]
+
+  @impl true
+  def init(attrs, ctx) do
+    fields = %{
+      "chart_type" => attrs["chart_type"] || "bar",
+      "width" => attrs["width"],
+      "height" => attrs["height"],
+      "x_field" => attrs["x_field"],
+      "y_field" => attrs["y_field"],
+      "x_field_type" => attrs["x_field_type"],
+      "y_field_type" => attrs["y_field_type"],
+      "color_field" => attrs["color_field"],
+      "color_field_type" => attrs["color_field_type"],
+      "data_variable" => attrs["data_variable"]
+    }
+
+    ctx =
+      assign(ctx,
+        fields: fields,
+        options: %{},
+        vl_alias: nil,
+        missing_dep: missing_dep(),
+        fresh: is_fresh?(attrs)
+      )
+
+    {:ok, ctx, reevaluate_on_change: true}
+  end
+
+  @impl true
+  def scan_binding(pid, binding, env) do
+    data_options =
+      for {key, val} <- binding,
+          is_atom(key),
+          is_map(val),
+          into: %{},
+          do: {Atom.to_string(key), Map.keys(val)}
```

Comment:
```suggestion
          do: {Atom.to_string(key), val |> Map.keys() |> Enum.map(&to_string/1)}
```

Reply:
Ah, in fact the map may have keys that we cannot convert to strings, we should filter only those that have atom/string keys then.

(in the future we will have a separate abstraction for tabular data that will be more precise and flexible, so for now we just don't want to crash ^^)

Reply:
`data = %{3 => [1, 2, 3], 4 => [1, 2, 3]}` this is working! Are there other cases?

Reply:
The problem is with terms that don't support `to_string`, like this:

```
data = %{self() => [1, 2, 3]}
```

Reply:
I see! I tried to solve it for now. Thanks!